datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -0,0 +1,270 @@
+            args=('target',),
+            doc="""Sibling name to create for this publication target.
+                If RECURSIVE is set, the same name will be used to address
+                the subdatasets' siblings""",
+            constraints=EnsureStr() | EnsureNone()),
+        sshurl=Parameter(
+            args=("sshurl",),
+            doc="""SSH URL to use to create the target sibling(s)""",
+            constraints=EnsureStr() | EnsureNone()),
+        target_dir=Parameter(
+            args=('--target-dir',),
+            doc="""Directory on the server where to create the repository and
+                that will be accessible via `target_url`. By
+                default it's wherever `sshurl` points to.
+                If you want to publish recursively, it is expected, that you
+                pass a template for building the URLs of all (sub)datasets to
+                be published by using placeholders.\n
+                List of currently available placeholders:\n
+                %%NAME\tthe name of the handle, where slashes are
+                replaced by dashes.\n""",
+            constraints=EnsureStr() | EnsureNone()),
+        target_url=Parameter(
+            args=('--target-url',),
+            doc="""The URL of the dataset sibling named by `target`. Defaults
+                to `sshurl`. This URL has to be accessible to anyone, who is
+                supposed to have access to the published dataset later on.\n
+                If you want to publish with `recursive`, it is expected, that
+                you pass a template for building the URLs of all (sub)datasets
+                to be published by using placeholders.\n
+                List of currently available placeholders:\n
+                %%NAME\tthe name of the dataset, where slashes are replaced by
+                dashes.\nThis option is ignored if there is already a
+                configured sibling dataset under the name given by `target`""",
+            nargs="?",
+            constraints=EnsureStr() | EnsureNone()),
+        target_pushurl=Parameter(
+            args=('--target-pushurl',),
+            doc="""Defaults to `sshurl`. In case the `target_url` cannot be
+                used to publish to the dataset sibling, this option specifies a
+                URL to be used for the actual publication operation.\n
+                This option is ignored if there is already a configured sibling
+                dataset under the name given by `target`""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=Parameter(
+            args=("--recursive", "-r"),
+            action="store_true",
+            doc="""Recursively create the publication target for all
+                subdatasets of `dataset`""",),
+        force=Parameter(
+            args=("--force", "-f",),
+            action="store_true",
+            doc="""If target directory exists already, force to (re-)init
+                git""",),)
```

Comment:
I've had the same thought, yes. Probably force should force everything. Reconfigure remote, in case it exists, reinit git. There is a related question: Remote exists, fetch url is the very same, but push url is either the same as fetch url (which might mean, that it wasn't explicitly set) or at least differs from what is given by --target-pushurl - do we need --force to set the new push url or do we do it anyway?